File: app/src/main/java/org/neidhardt/dynamicsoundboard/ApplicationComponent.java
package org.neidhardt.dynamicsoundboard;

import dagger.Component;
import org.neidhardt.dynamicsoundboard.navigationdrawer.NavigationDrawerFragment;
import org.neidhardt.dynamicsoundboard.navigationdrawer.playlist.views.Playlist;
import org.neidhardt.dynamicsoundboard.navigationdrawer.soundsheets.views.SoundSheets;
import org.neidhardt.dynamicsoundboard.soundactivity.SoundActivity;
import org.neidhardt.dynamicsoundboard.soundcontrol.PauseSoundOnCallListener;
import org.neidhardt.dynamicsoundboard.soundcontrol.SoundSheetFragment;
import org.neidhardt.dynamicsoundboard.soundmanagement.dagger.SoundsDataModule;
import org.neidhardt.dynamicsoundboard.notifications.service.NotificationService;
import org.neidhardt.dynamicsoundboard.soundmanagement.model.SoundsDataAccess;
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.dagger.SoundSheetsDataModule;
import org.neidhardt.dynamicsoundboard.views.BaseDialog;

import javax.inject.Singleton;

/**
 * File created by eric.neidhardt on 16.06.2015.
 */
@Singleton
@Component(modules = {SoundsDataModule.class, SoundSheetsDataModule.class})
public interface ApplicationComponent
{
	void inject(NotificationService service);

	void inject(BaseDialog dialog);

	void inject(SoundActivity activity);

	void inject(Playlist playlist);

	void inject(SoundSheets soundSheets);

	void inject(SoundSheetFragment fragment);

	void inject(NavigationDrawerFragment fragment);

	void inject(PauseSoundOnCallListener listener);

	SoundsDataAccess provideSoundsDataAccess();
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/ApplicationComponent.kt
package org.neidhardt.dynamicsoundboard

import org.neidhardt.dynamicsoundboard.soundmanagement.model.SoundsDataAccess
import org.neidhardt.dynamicsoundboard.soundmanagement.model.SoundsDataStorage
import org.neidhardt.dynamicsoundboard.soundmanagement.model.SoundsDataUtil
import org.neidhardt.dynamicsoundboard.soundmanagement.model.SoundsManager
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsDataAccess
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsDataStorage
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsDataUtil
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsManager

/**
 * File created by eric.neidhardt on 09.07.2015.
 */
public class ApplicationComponent()
{
	private val soundsManager = SoundsManager()
	private val soundSheetsManager = SoundSheetsManager()

	public val soundsDataAccess: SoundsDataAccess = this.soundsManager
	public val soundsDataStorage: SoundsDataStorage = this.soundsManager
	public val soundsDataUtil: SoundsDataUtil = this.soundsManager

	public val soundSheetsDataAccess: SoundSheetsDataAccess = this.soundSheetsManager
	public val soundSheetsDataStorage: SoundSheetsDataStorage = this.soundSheetsManager
	public val soundSheetsDataUtil: SoundSheetsDataUtil = this.soundSheetsManager
}